### PR TITLE
Create Buttons Resources,  Back To Search, and Show Route Buttons

### PR DESCRIPTION
**Pull request by RCAVelez**
**What's new:**
Created Buttons folder, buttons js and buttons css which will contain all functions for buttons and css.
Created back to search button which routes to /
Created show route button which routes to /route
Created back to search and show route button js functions and css 

**What's changed:**
Changed double quotes to single quotes in js files

### CODE REVIEW

File: frontend/src/index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import App from './App';
 import * as serviceWorker from './serviceWorker';
 import './App.css';
 import ListImagesComponent from './ListImagesComponent/ListImagesComponent'
+import {BackButton,RouteButton} from './Buttons/Buttons'
```
Comment (mecolmg) [inline frontend/src/index.js:8]: Not using this, please remove.

File: frontend/src/Buttons/Buttons.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+function BackButton() {
```
Comment (mecolmg) [inline frontend/src/Buttons/Buttons.js:4]: Add JsDocs for all exported components/methods/vars/etc.

File: frontend/src/App.css [hunk 3]
```diff
@@ -1,5 +1,6 @@
 @import "./ListImagesComponent/ListImagesComponent.css";
 @import "./ImageComponent/ImageComponent.css";
+@import "./Buttons/Buttons.css";
```
Comment (mecolmg) [inline frontend/src/App.css:3]: As mentioned in the other PRs, don't import this here. Import in the JS file where it's used.

File: frontend/src/Buttons/Buttons.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#backToSearchButton {
+  min-height: 50px;
+  min-width: 200px;
+  padding: 10px;
+}
+
+#backToSearchButton:hover {
+  opacity: 0.5;
+}
```
Comment (mecolmg) [inline frontend/src/Buttons/Buttons.css:9]: This is identical to below, so don't duplicate it. Just create a shared class between the two buttons.

File: frontend/src/Buttons/Buttons.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (mecolmg) [inline frontend/src/Buttons/Buttons.js:1]: Imagine how many buttons a single application can have. It doesn't make sense to have a single file to implement all buttons. Also, in general it's a best practice to have only one exported component per file. 

To take advantage of components, you should instead make this a NavButton component which takes in a link and label prop.

File: frontend/src/Buttons/Buttons.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import PropTypes from 'prop-types'
```
Comment (genedwards) [inline frontend/src/Buttons/Buttons.js:2]: unused import here
The linter would've caught this, please merge in master to this branch (or rebase onto master) and run the linter.

File: frontend/src/Buttons/Buttons.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+function BackButton() {
+  return (
+    <div className='backToSearchButtonContainer'>
+      <a href='/'>
+        <button id='backToSearchButton' type='button' class='btn btn-primary'>Back</button>
```
Comment (Cindyzhang977) [inline frontend/src/Buttons/Buttons.js:8]: I think React uses `className` instead of `class`

File: frontend/src/ImageComponent/ImageComponent.js [hunk 8]
```diff
@@ -2,11 +2,11 @@ import React from 'react'
 import PropTypes from 'prop-types'
```
Comment (Cindyzhang977) [inline frontend/src/ImageComponent/ImageComponent.js:2]: Is this used?

File: frontend/src/index.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ import App from './App';
...
 
 ReactDOM.render(
   <React.StrictMode>
```
Comment (Cindyzhang977) [inline frontend/src/index.js:11]: What's StrictMode?

File: frontend/src/index.js (hunk 1 above)
Comment (RCAVelez) [reply frontend/src/index.js:8]: done

File: frontend/src/Buttons/Buttons.js (hunk 2 above)
Comment (RCAVelez) [reply frontend/src/Buttons/Buttons.js:4]: done

File: frontend/src/App.css (hunk 3 above)
Comment (RCAVelez) [reply frontend/src/App.css:3]: done

File: frontend/src/Buttons/Buttons.css (hunk 4 above)
Comment (RCAVelez) [reply frontend/src/Buttons/Buttons.css:9]: changed to nav-button and removed duplicates

File: frontend/src/Buttons/Buttons.js (hunk 5 above)
Comment (RCAVelez) [reply frontend/src/Buttons/Buttons.js:1]: perfect. Buttons is not named nav-button. It exports only one component and it takes in 2 properties. link and label

File: frontend/src/Buttons/Buttons.js (hunk 6 above)
Comment (RCAVelez) [reply frontend/src/Buttons/Buttons.js:2]: gone! I'll rebase to master soon to have the linter

File: frontend/src/Buttons/Buttons.js (hunk 7 above)
Comment (RCAVelez) [reply frontend/src/Buttons/Buttons.js:8]: Hm, it still styled. Ok! Changed to className, still works! Done

File: frontend/src/ImageComponent/ImageComponent.js (hunk 8 above)
Comment (RCAVelez) [reply frontend/src/ImageComponent/ImageComponent.js:2]: casted to shadow realm!

File: frontend/src/index.js (hunk 9 above)
Comment (RCAVelez) [reply frontend/src/index.js:11]: idk what strictMode is. It was there when I pulled from master yesterday but it does the following
StrictMode currently helps with:

Identifying components with unsafe lifecycles
Warning about legacy string ref API usage
Warning about deprecated findDOMNode usage
Detecting unexpected side effects
Detecting legacy context API

[StrictMode](https://reactjs.org/docs/strict-mode.html)

File: frontend/src/buttons/nav-button.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * NavButton is passed two properties
+ * url which has the link to go to
+ * text which has the button label
+ **/
+function NavButton(props) {
+  return (
+    <a href={props.link}>
+      <button id='nav-button' type='button' className='btn btn-primary'>{props.label}</button>
```
Comment (Cindyzhang977) [inline frontend/src/buttons/nav-button.js:12]: Just to be consistent, use the React Bootstrap Button: https://react-bootstrap.github.io/components/buttons/

I think in this case it would be 
```
      <Button id="nav-button" variant="primary">{props.label}</Button>
```

(also make sure the html uses double quotes)

File: frontend/src/attractions-panel/attractions-panel.js [hunk 11]
```diff
@@ -6,12 +6,12 @@ import './attractions-panel.css';
...
-function AttractionsPanel(attractionsUrls) {
+function AttractionsPanel(props) {
   return (
     <div className="attractions-panel">
-      <Attraction imageUrl={attractionsUrls.attractionsUrls[0]} />
-      <Attraction imageUrl={attractionsUrls.attractionsUrls[1]} />
-      <Attraction imageUrl={attractionsUrls.attractionsUrls[2]} />
+      <Attraction imageUrl={props.attractionsUrls[0]} />
+      <Attraction imageUrl={props.attractionsUrls[1]} />
+      <Attraction imageUrl={props.attractionsUrls[2]} />
```
Comment (Cindyzhang977) [inline frontend/src/attractions-panel/attractions-panel.js:14]: If you want to make it clearer you can write
```
function AttractionsPanel({attractionsUrls}) 
```
and then do 
```
<Attraction imageUrl={attractionsUrls[0]} />
```

File: frontend/src/buttons/nav-button.js (hunk 10 above)
Comment (genedwards) [reply frontend/src/buttons/nav-button.js:12]: Prettier will change single quotes to double quotes automatically.

Comment (mecolmg) [inline frontend/src/buttons/nav-button.js:12]: Generally ids should only be used once per page. In my experience you should never actually need ids for CSS, instead use a class.

File: frontend/src/buttons/nav-button.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import './nav-button.css';
+
+/**
+ * NavButton is passed two properties
+ * url which has the link to go to
+ * text which has the button label
+ **/
```
Comment (mecolmg) [inline frontend/src/buttons/nav-button.js:8]: nit: The end of a block comment doesn't need two `*`. 

Also, the JsDoc should primarily describe what the component is (i.e. `Navigation bar link button.`).

As for the inputs to the component, there's several ways to document this. The easiest first step is to deconstruct your props like:
```js
function NavButton({text, url})
```
If you want to go further, you can use [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html).

File: frontend/src/attractions-panel/attractions-panel.js [hunk 13]
```diff
@@ -6,12 +6,12 @@ import './attractions-panel.css';
   * AttractionsPanel shows all attractions.
   * Updates browser url when an image selection is toggled
 **/
-function AttractionsPanel(attractionsUrls) {
+function AttractionsPanel(props) {
```
Comment (mecolmg) [inline frontend/src/attractions-panel/attractions-panel.js:9]: This doesn't belong in this PR, it should be in the previous one.

File: frontend/src/buttons/nav-button.css [hunk 14]
```diff
@@ -0,0 +1,9 @@
+#nav-button {
```
Comment (mecolmg) [inline frontend/src/buttons/nav-button.css:1]: Same as previous PRs, this file should be PascalCase (`NavButton.css`). Also, the directory it's written in should be more specific than `buttons`, consider `navigation`.

File: frontend/src/buttons/nav-button.js (hunk 10 above)
Comment (mecolmg) [reply frontend/src/buttons/nav-button.js:12]: Even better, the Bootstrap button component takes in a `href` prop, so you can remove the `<a>` tag surrounding the button.

https://react-bootstrap.github.io/components/buttons/#button-props

File: frontend/src/attractions-panel/attractions-panel.js (hunk 11 above)
Comment (RCAVelez) [reply frontend/src/attractions-panel/attractions-panel.js:14]: done!

File: frontend/src/buttons/nav-button.js (hunk 10 above)
Comment (RCAVelez) [reply frontend/src/buttons/nav-button.js:12]: done!

Comment (RCAVelez) [reply frontend/src/buttons/nav-button.js:12]: got it changed to class

File: frontend/src/buttons/nav-button.js (hunk 12 above)
Comment (RCAVelez) [reply frontend/src/buttons/nav-button.js:8]: got it, I explained simply what it is and I deconstructed the props

File: frontend/src/attractions-panel/attractions-panel.js (hunk 13 above)
Comment (RCAVelez) [reply frontend/src/attractions-panel/attractions-panel.js:9]: I'm not sure how that happened. I'm going to push again and see if it disappears

File: frontend/src/buttons/nav-button.css (hunk 14 above)
Comment (RCAVelez) [reply frontend/src/buttons/nav-button.css:1]: done

File: frontend/src/navigation/NavButton.css [hunk 15]
```diff
@@ -0,0 +1,9 @@
+.nav-button {
```
Comment (mecolmg) [inline frontend/src/navigation/NavButton.css:1]: Update to .module.css file, and change class names to camelCase as discussed in chat.

Comment (RCAVelez) [reply frontend/src/navigation/NavButton.css]: done

File: frontend/src/navigation/NavButton.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from './NavButton.module.css';
+import Button from 'react-bootstrap/Button';
+
+/**
+ * Navigation bar link button
+ */
+function NavButton({link,label}) {
+  return (
+    <Button href={link} type='button' className={styles['navButton']}>{label}</Button>
```
Comment (zoebarth) [inline frontend/src/navigation/NavButton.js:10]: Is there a reason for using `className={styles['navButton']}` as opposed to `className={styles.navButton}`?

Comment (genedwards) [reply frontend/src/navigation/NavButton.js:10]: https://github.com/css-modules/css-modules#naming recommends {styles.className}, we should follow that.

Comment (RCAVelez) [reply frontend/src/navigation/NavButton.js:10]: ok will follow done

File: frontend/src/navigation/NavButton.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from './NavButton.module.css';
+import Button from 'react-bootstrap/Button';
+
+/**
+ * Navigation bar link button
+ */
+function NavButton({link,label}) {
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/navigation/NavButton.js:8]: done
